src/components/RegionSettings.tsx

```diff
@@ -24,7 +24,7 @@ interface Region {
 
 interface RegionSettingsProps {
   regions: Array<Region>;
-  selectableRegions: Array<string>;
+  selectableRegions: Array<string | undefined>;
```

Comment:
Dlaczego chcesz przechowywać listę z `undefined` w stanie?

Według mnie to nie jest potrzebne, bo nie będziemy w stanie wybrać taki region, a oprócz tego utrudnia funkcjonowanie na nim ponieważ za każdym razem potrzeba będzie sprawdzać czy wartość jest undefindem